rest/server_context.go

```diff
@@ -70,6 +70,7 @@ type ServerContext struct {
 	LogContextID                  string          // ID to differentiate log messages from different server context
 	fetchConfigsLastUpdate        time.Time       // The last time fetchConfigsWithTTL() updated dbConfigs
 	allowScopesInPersistentConfig bool            // Test only backdoor to allow scopes in persistent config, not supported for multiple databases with different collections targeting the same bucket
+	activeReplicatorCount         int             // The count of concurrent active replicators
```

Comment:
if we make this an `atomic.Int32` is that going be better than accessing requiring a full context lock, especially that this is going to update frequently? The golang 1.19 atomic types are pretty great because the compiler will complain if you don't access them without loading.

Reply:
I did think about this but ideally i'd want to lock the server context so the max concurrent replicatiosn value doesn't get changed by an incoming request during execution so thought it would be best to lock. Although I understand there is some performance degredation when locking and unlocking frequently in comparison to using atomic.